docs/plans/2026-02-21-modifications-db-separation-of-concerns.md

```
@@ -0,0 +1,1226 @@
+Update includes:
+
+```cpp
+#include <OpenMS/CHEMISTRY/CrossLinksDB.h>
+#include <OpenMS/CHEMISTRY/OBODataProvider.h>
+```
+
+Remove includes that are no longer needed: `<OpenMS/SYSTEM/File.h>`, `<fstream>`.
+
+The destructor stays as-is (the parent ModificationsDB destructor handles cleanup).
+
+Actually, looking at CrossLinksDB's destructor — it manually deletes mods_ entries. But the parent destructor does the same thing. This means mods get double-deleted. Check whether the parent destructor is virtual (it is: `virtual ~ModificationsDB()`). If CrossLinksDB's destructor runs first and clears mods_, the parent destructor would iterate an empty vector, which is safe. Keep CrossLinksDB's destructor as-is to match existing behavior.
```

markdownlint-cli2 (0.21.0)

[warning] Spaces inside emphasis markers

(MD037, no-space-in-emphasis)